backend/src/main/kotlin/com/infowings/catalog/data/objekt/ObjectService.kt

```diff
@@ -194,22 +244,60 @@ class ObjectService(
             }
             historyService.storeFact(valueVertex.toCreateFact(context))
 
-            return@transaction valueVertex.toObjectPropertyValue()
+            ValueResult(
+                valueVertex,
+                valueVertex.toObjectPropertyValue().value.toObjectValueData().toDTO(),
+                valueVertex.measure?.id,
+                valueVertex.objectProperty ?: throw IllegalStateException("Object value was created without reference to object property"),
+                valueVertex.aspectProperty,
+                valueVertex.parentValue
+            )
         }
+
+        return ValueCreateResponse(
+            valueCreateResult.id,
+            valueCreateResult.valueDto,
+            valueCreateResult.description,
+            valueCreateResult.measureId,
+            Reference(valueCreateResult.objectPropertyId, valueCreateResult.objectPropertyVersion),
+            valueCreateResult.aspectPropertyId,
+            valueCreateResult.parentValueId?.let { id -> valueCreateResult.parentValueVersion?.let { version -> Reference(id, version) } },
+            valueCreateResult.version
+        )
     }
 
-    fun update(request: ValueUpdateRequest, username: String): ObjectPropertyValue {
+    fun update(request: ValueUpdateRequest, username: String): ValueUpdateResponse {
         val userVertex = userService.findUserVertexByUsername(username)
         val context = HistoryContext(userVertex)
-        return transaction(db) {
-            var objectPropertyValue = findPropertyValueById(request.valueId)
-            val valueInfo: ValueWriteInfo = validator.checkedForUpdating(objectPropertyValue, request)
-            val before = objectPropertyValue.currentSnapshot()
-            objectPropertyValue = dao.saveObjectValue(objectPropertyValue, valueInfo)
-            historyService.storeFact(objectPropertyValue.toUpdateFact(context, before))
-
-            return@transaction objectPropertyValue.toObjectPropertyValue()
+        val valueUpdateResult = transaction(db) {
+            var valueVertex = findPropertyValueById(request.valueId)
+            val valueInfo: ValueWriteInfo = validator.checkedForUpdating(valueVertex, request)
+            val before = valueVertex.currentSnapshot()
+
+            valueVertex = dao.saveObjectValue(valueVertex, valueInfo)
+
+            historyService.storeFact(valueVertex.toUpdateFact(context, before))
+
+            ValueResult(
```

Comment:
A lot of code duplication here. Maybe extract some extension function like `valueVertex.toValueResult()`

Reply:
I'll look into it, will try to refactor. As I remember there was a problem with retrieving entities after delete. But I will try to make it look better

Reply:
done